src/Spark.Engine/Extensions/SearchParamDefinitionExtensions.cs

```diff
@@ -23,5 +23,32 @@ internal static bool CanHaveOperatorPrefix(this List<SearchParamDefinition> sear
                 || searchParamDefinition.Type == SearchParamType.Date
                 || searchParamDefinition.Type == SearchParamType.Quantity);
         }
+        
+        /// <summary>
+        /// Returns true if the search parameter is one of the following types: Number, Date or Quantity.
+        /// See https://www.hl7.org/fhir/stu3/search.html#prefix for more information.
+        /// </summary>
+        /// <param name="searchParamDefinitions">
+        /// A List of <see cref="SearchParamDefinition"/>, since this is an extension this is usually a reference 
+        /// to ModelInfo.SearcParameters.
+        /// </param>
+        /// <param name="resourceType">A <see cref="string"/> representing the resource type of the search parameter</param>
+        /// <param name="name">A <see cref="string"/> representing the name of the search parameter.</param>
+        /// <returns>Returns true if the search parameter is of type Number, Date or Quanity, otherwise false.</returns>
+        internal static bool CanHaveOperatorPrefix(this List<SearchParamDefinition> searchParamDefinitions, string resourceType, string name)
+        {
+            // If this is a global SearchParameter then do not include the resource type.
+            SearchParamDefinition searchParamDefinition = IsGlobalSearchParameter(name)
+                ? searchParamDefinitions.Find(p => p.Name == name)
+                : searchParamDefinitions.Find(p => p.Resource == resourceType && p.Name == name);
+            return searchParamDefinition != null && (searchParamDefinition.Type == SearchParamType.Number
+                                                     || searchParamDefinition.Type == SearchParamType.Date
+                                                     || searchParamDefinition.Type == SearchParamType.Quantity);
+        }
+
+        private static bool IsGlobalSearchParameter(string name)
+        {
+            return name.IndexOf('_') == 0;
```

Comment:
@kennethmyhra  Can the IsGlobalSearchParameter can be judged with `p.Resource == nameof(Resource)`?

Reply:
@whyfate That could be, do you have a chance to try it out?

Reply:
This is my code.
```csharp
internal static bool CanHaveOperatorPrefix(this List<SearchParamDefinition> searchParamDefinitions, string resourceType, string name)
        {
            SearchParamDefinition searchParamDefinition = searchParamDefinitions.Find(p => (p.Resource == resourceType || p.Resource == nameof(Resource)) && p.Name == name);
            return searchParamDefinition != null && (searchParamDefinition.Type == SearchParamType.Number
                || searchParamDefinition.Type == SearchParamType.Date
                || searchParamDefinition.Type == SearchParamType.Quantity);
        }

```
I don't know which is better for understanding the code.

Reply:
I think yours is a better and more proper implementation. Can you give me a PR with those changes so the integration tests can give it a run.

Reply:
Yes,I can try it.